### PR TITLE
Integrate i18n material from CLI wiki

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

Incorporates material from  https://github.com/angular/angular-cli/wiki/stories-internationalization
into existing i18n guide.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: aio/content/guide/i18n.md [hunk 1]
```diff
@@ -10,14 +10,21 @@ an AOT-compiled app, translated into French.
...
 * Handling plural forms of words.
 * Handling alternative text.
 
-This document focuses on [**Angular CLI**](https://cli.angular.io/) projects, in which the Angular
-CLI generates most of the boilerplate necessary to write your app in multiple languages.
+For localization, you can use the [**Angular CLI**](https://cli.angular.io/) to generate most of the boilerplate necessary to create files for translators, and to publish it in multiple languages. 
+After you have set up your app to use i18n, the CLI can help you with the following steps:
+* Extracting localizable text into a file that you can send out to be translated.
+* Building and serving the app for a given locale, using the translated text.
+* Switching among multiple languages.
```
Comment (ocombe) [inline aio/content/guide/i18n.md]: I think that this is misleading.
You can't switch the language of your app at runtime, you can build your app for multiple languages [with the CLI], and then it's up to you to find a way to load each version

File: aio/content/guide/i18n.md [hunk 2]
```diff
@@ -129,13 +136,13 @@ The i18n template translation process has four phases:
 
 1. Mark static text messages in your component templates for translation.
 
-2. An Angular i18n tool extracts the marked text into an industry standard translation source file.
+2. Create translation files: use the Angular CLI `i18n` tool to extract the marked text into an industry-standard translation source file.
 
-3. A translator edits that file, translating the extracted text into the target language,
-and returns the file to you.
+3. Send the translation file to a translator. The translator edits that file, translating the extracted text into the target language, and returns the file to you.
```
Comment (ocombe) [inline aio/content/guide/i18n.md]: I don't think this step adds any value, a simple "translate the file into the target language" would probably be better (people are free to use an external translator or not)

File: aio/content/guide/i18n.md [hunk 3]
```diff
@@ -281,16 +288,14 @@ The `<ng-container>` is transformed into an html comment:
 </code-example>
 
 {@a translate-attributes}
-## Add i18n translation attributes
+### Translating attribute text
```
Comment (ocombe) [inline aio/content/guide/i18n.md]: "attributes" ? if you're translating attributes, there's a good chance that you won't translate only one

File: aio/content/guide/i18n.md [hunk 4]
```diff
@@ -303,10 +308,14 @@ This technique works for any attribute of any element.
 You also can assign a meaning, description, and id with the `i18n-x="<meaning>|<description>@@<id>"`
 syntax.
 
-{@a plural-ICU}
-## Translate singular and plural
+## Regular expressions for plurals and selections
 
-Different languages have different pluralization rules.
+Different languages have different pluralization rules and grammatical constructions that add 
+complexity to the translation task. The [International Components for Unicode (ICU) library](http://userguide.icu-project.org) provides a [Regular Expressions package](http://userguide.icu-project.org/strings/regexp) that lets you apply regular expression matching to Unicode string data.
```
Comment (ocombe) [inline aio/content/guide/i18n.md]: Where does this come from? we don't use ICU regexps in Angular

File: aio/content/guide/i18n.md [hunk 5]
```diff
@@ -359,7 +368,7 @@ for two, three, or any other number if the pluralization rules were different. F
 </div>
 
 {@a select-ICU}
-## Select among alternative text messages
+### Selection among alternative text messages
```
Comment (ocombe) [inline aio/content/guide/i18n.md]: Are you sure about this change? It sounds weird

**Review by ocombe — CHANGES_REQUESTED** [review]

**Comment by jbogarthyde** [comment]
Thanks @ocombe -- please check changes and approve or correct.

File: aio/content/guide/i18n.md [hunk 6]
```diff
@@ -311,8 +311,8 @@ syntax.
 ## Regular expressions for plurals and selections
 
 Different languages have different pluralization rules and grammatical constructions that add 
-complexity to the translation task. The [International Components for Unicode (ICU) library](http://userguide.icu-project.org) provides a [Regular Expressions package](http://userguide.icu-project.org/strings/regexp) that lets you apply regular expression matching to Unicode string data.
-
+complexity to the translation task. 
+Use regular expressions with the `plural` and `select` clauses to provide patterns that aid translation in these cases.
```
Comment (ocombe) [inline aio/content/guide/i18n.md]: I still don't know where this paragraph comes from (title above and those 2 sentences)? And I don't know what "Use regular expressions with the `plural` and `select` clauses to provide patterns that aid translation in these cases." means without an example. Why was this added?

File: aio/content/guide/i18n.md [hunk 7]
```diff
@@ -322,7 +322,7 @@ In English, depending upon the number of minutes, you could display "just now",
 or "x minutes ago" (with x being the actual number).
 Other languages might express the cardinality differently.
 
-The example below shows how to use a `plural` ICU expression to display one of those three options
+The example below shows how to use a `plural` expression to display one of those three options
```
Comment (ocombe) [inline aio/content/guide/i18n.md]: I think that "`plural` ICU expression" was correct

File: aio/content/guide/i18n.md [hunk 8]
```diff
@@ -368,12 +368,12 @@ for two, three, or any other number if the pluralization rules were different. F
...
 
 {@a select-ICU}
-### Selection among alternative text messages
+### Select among alternative text messages
 
 If your template needs to display different text messages depending on the value of a variable, you
 need to translate all of those alternative text messages.
 
-You can handle this with a `select` ICU expression. It is similar to the `plural` ICU expressions
+You can handle this with a `select` expression. It is similar to the `plural` expressions
```
Comment (ocombe) [inline aio/content/guide/i18n.md]: same comment as above, I think that you should keep "ICU" here

**Review by ocombe — CHANGES_REQUESTED** [review]

File: aio/content/guide/i18n.md (hunk 6 above)
Comment (jbogarthyde) [reply aio/content/guide/i18n.md]: This is meant for organization - I want to introduce how the `plural` and `select` concepts are related conceptually, which explains why we are discussing them together. Can you think of a better way to describe how they are related? 
Also, since we do mention ICU expressions, it does seem like we should include the pointer to what that means. I'd like to put those links back, if you can help me understand how it fits in.

**Review by ocombe — CHANGES_REQUESTED** [review]

**Comment by jbogarthyde** [comment]
@ocombe  I've removed the link to the ICU page that doesn't apply to JavaScript/TypeScript. 
If you know of anyplace that discusses the regular-expression syntax that can be used here, I could put that in.

**Review by ocombe — APPROVED** [review]

**Comment by jbogarthyde** [comment]
@jenniferfell Needs pull-approve

**Review by jenniferfell — CHANGES_REQUESTED** [review]
Lots of good changes. The one thing this document can help us standardize is how we talk about CLI commands in the context of other procedures/activities. 

the Angular CLI command `xi18n`
the CLI command `xi18n`
the `xi18n` tool 
the `ng xi81n` command
the CLI command `ng xi18n` 
the Angular CLI command `ng xi18n`

Maybe "the Angular" only in first use and then something shorter later?

File: aio/content/guide/i18n.md [hunk 9]
```diff
@@ -129,13 +136,13 @@ The i18n template translation process has four phases:
 
 1. Mark static text messages in your component templates for translation.
 
-2. An Angular i18n tool extracts the marked text into an industry standard translation source file.
+2. Create translation files: use the Angular CLI `i18n` tool to extract the marked text into an industry-standard translation source file.
```
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/guide/i18n.md]: I would say the `i18n` tool is what the CLI invokes with the `xi18n' command -- but I'm changing "`i18n` tool" to "`xi18n` command" to avoid confusion.

File: aio/content/guide/i18n.md [hunk 10]
```diff
@@ -129,13 +136,13 @@ The i18n template translation process has four phases:
...
 
-3. A translator edits that file, translating the extracted text into the target language,
-and returns the file to you.
+3. Edit the translation files to translate the extracted text into the target language.
 
-4. The Angular compiler imports the completed translation files,
-replaces the original messages with translated text, and generates a new version of the app
+4. Merge the completed translation file into the app. 
+Use the Angular CLI `build` tool to compile the app, choosing compiler options to import the completed translation files,
```
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply aio/content/guide/i18n.md]: We need to edit the existing compiler page to agree with (or perhaps just point to) the new CLI ref for `build` options, and update all of these with correct xrefs when the pages land.

File: aio/content/guide/i18n.md [hunk 11]
```diff
@@ -129,14 +136,17 @@ The i18n template translation process has four phases:
...
-2. An Angular i18n tool extracts the marked text into an industry standard translation source file.
+2. Create a translation file: Use the Angular CLI `xi18n` command to extract the marked text into an industry-standard translation source file.
+
+3. Edit the generated translation file: Translate the extracted text into the target language.
+
+4. Merge the completed translation file into the app. To do this, use the Angular CLI `build` command to compile the app, choosing a [locale-specific configuration](#merge-aot), or specifying the following command options.
 
-3. A translator edits that file, translating the extracted text into the target language,
-and returns the file to you.
+* `--i18nFile`=*path to the translation file*
```
Comment (jenniferfell) [inline aio/content/guide/i18n.md]: To indent sublist in markdown, add 4 blank spaces or a tab at start of the line: 
https://commonmark.org/help/tutorial/10-nestedLists.html
https://github.com/tchapi/markdown-cheatsheet/blob/master/README.md

**Review by jenniferfell — APPROVED** [review]
One markdown tip if you want, but not a condition of merge.
